NetCoreRestApi/DataLayer/UnitOfWorks/CategoryUnitOfWork.cs

```diff
@@ -0,0 +1,48 @@
+﻿using DataLayer.Entities;
+using DataLayer.Repositories;
+using DataLayer.UnitOfWorks.Interfaces;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace DataLayer.UnitOfWorks
+{
+    public class CategoryUnitOfWork<TId> : ICategoryUnitOfWork<TId>
+    {
+        private ICategoryRepository<TId> _categoryRepository;
+
+        public CategoryUnitOfWork(ICategoryRepository<TId> categoryRepository)
+        {
+            _categoryRepository = categoryRepository;
+        }
+
+        public CategoryEntity<TId> GetById(TId id)
```

Comment:
Move Generic implementations to BaseUnitOfWork

Reply:
done